node-wasm/src/ports.rs

```diff
@@ -0,0 +1,314 @@
+use js_sys::{Array, Function, Reflect};
+use serde::Serialize;
+use serde_wasm_bindgen::{from_value, to_value, Serializer};
+use tokio::select;
+use tokio::sync::{mpsc, Mutex};
+use tracing::{error, info, trace};
+use wasm_bindgen::closure::Closure;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::{JsCast, JsValue};
+use wasm_bindgen_futures::spawn_local;
+use web_sys::{MessageEvent, MessagePort};
+
+use crate::commands::{NodeCommand, WorkerResponse};
+use crate::error::{Context, Error, Result};
+
+// Instead of just supporting communicaton with just `MessagePort`, allow using any object which
+// provides compatible interface
```

Comment:
```suggestion
// Instead of supporting communication with just `MessagePort`, allow using any object which
// provides compatible interface, eg. `Worker`
```